src/main/java/com/crowdin/cli/utils/ConcurrencyUtil.java

```diff
@@ -24,15 +23,22 @@ public static void executeAndWait(List<Runnable> tasks) {
             return;
         }
         ExecutorService executor = Executors.newFixedThreadPool(CROWDIN_API_MAX_CONCURRENT_REQUESTS);
-        tasks.forEach(executor::submit);
-        executor.shutdown();
+        List<Future<?>> futures = tasks
+            .stream()
+            .map(executor::submit)
+            .collect(Collectors.toList());
         try {
-            if (!executor.awaitTermination(tasks.size() * 2, TimeUnit.MINUTES)) {
-                executor.shutdownNow();
+            for(Future<?> future : futures) {
+                try {
+                    future.get(2, TimeUnit.MINUTES);
```

Comment:
Not sure if this is good solution because we have loop where in each iteration we invoking blocking code. It might cause performance degradation. Here is nice [solution](https://stackoverflow.com/a/2248203) which will do the same thing but inside of executor which therefore will execute blocking code `future.get` in separate working thread.

Reply:
Yeah, thanks for the better solution.